apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowFilters.cs

```diff
@@ -0,0 +1,380 @@
+        /// columns whose qualifiers satisfy the given RE2 regex.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note that, since column qualifiers can contain arbitrary bytes, the `\C`
+        /// escape sequence must be used if a true wildcard is desired. The `.`
+        /// character will not match the new line character `\n`, which may be
+        /// present in a binary qualifier.
+        /// </para>
+        /// <para>
+        /// Note that string is implicitly convertible to <see cref="BigtableByteString"/>,
+        /// so <paramref name="regex"/> can be specified using a string as well and its UTF-8
+        /// representations will be used.
+        /// </para>
+        /// </remarks>
+        /// <param name="regex">
+        /// The RE2 regex used to match column qualifiers.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter ColumnQualifierRegex(BigtableByteString regex) =>
+            new RowFilter { ColumnQualifierRegexFilter = regex.Value };
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which matches only cells from
+        /// columns within the given range.
+        /// </summary>
+        /// <param name="range">
+        /// The range of columns from which cells should be matched. Must not be null.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter ColumnRange(ColumnRange range) =>
+            new RowFilter { ColumnRangeFilter = GaxPreconditions.CheckNotNull(range, nameof(range)) };
+
+        // TODO: Add additional ColumnRange... methods to take start/end values?
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which applies one of two
+        /// possible RowFilters to the data based on the output of a predicate
+        /// RowFilter.
+        /// </summary>
+        /// <param name="predicateFilter">
+        /// If <paramref name="predicateFilter"/> outputs any cells, then
+        /// <paramref name="trueFilter"/> will be evaluated on the input row.
+        /// Otherwise, <paramref name="falseFilter"/> will be evaluated.
+        /// Must not be null.
+        /// </param>
+        /// <param name="trueFilter">
+        /// The filter to apply to the input row if <paramref name="predicateFilter"/>
+        /// returns any results. If null, no results will be returned in the true case.
+        /// </param>
+        /// <param name="falseFilter">
+        /// The filter to apply to the input row if <paramref name="predicateFilter"/>
+        /// does not return any results. If null, no results will be returned in the
+        /// false case.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter Condition(
+            RowFilter predicateFilter, RowFilter trueFilter, RowFilter falseFilter) =>
```

Comment:
Validate that at least one of trueFilter or falseFilter is non-null?

Reply:
While it would be odd to not set either one, it isn't documented as not allowed. Again I'd like to err on the side of not imposing restrictions I assume make sense. But if you think it's best to do it anyway, I will.

Reply:
I think I got confused between this and CreateCheckAndMutateRowRequest that checks that there's at least one mutation.

Reply:
Oh I see. Yes, that is documented that you need at least one true/false mutation.